### ISSUE TITLE
Unwanted logs in output MacOs: [check_gcp_environment.cc:61] BIOS data file does not exist or cannot be opened.

### ISSUE THREAD

**Issue by sshishov** [grpc/grpc#37447]
### What version of gRPC and what language are you using?
Library version:`1.65.4`

### What operating system (Linux, Windows,...) and version?
OS: `MacOS M1 Pro (docker)`
Version: `Sonoma 14.5 (23F79)`

### What runtime / compiler are you using (e.g. python version or version of gcc)
Language: `Python`
Version `3.11.9`

### What did you do?
We are not using the library direclty, instead we are using:
```
google-cloud-bigquery
google-cloud-core
google-cloud-documentai
google-cloud-kms
google-cloud-storage
google-cloud-tasks
```

### What did you expect to see?
We would like to see no logs regarding the library we are not configured and used directly

### What did you see instead?

When we start the application, we see these logs on start:
```
WARNING: All log messages before absl::InitializeLog() is called are written to STDERR
I0000 00:00:1723277982.711175   15154 check_gcp_environment.cc:61] BIOS data file does not exist or cannot be opened.
```

### Anything else we should know about your project / environment?
- we are using Django + wsgi
- no logs appearing on docker under Ubuntu

**Comment by sourabhsinghs** [comment]
@tanvi-jagtap  Can you look into this.

**Comment by sebastian-correa** [comment]
Linked with #37178? I see a similar log in `1.65.4` on Python 3.10 and MacOS (Apple silicon):

```log
WARNING: All log messages before absl::InitializeLog() is called are written to STDERR
I0000 00:00:1723666552.540596 44436952 check_gcp_environment_no_op.cc:29] ALTS: Platforms other than Linux and Windows are not supported
```

**Comment by sebastian-correa** [comment]
@sshishov did you try 1.65.5? It seems to have fixed this for us.

**Comment by sshishov** [comment]
I can reproduce the same log with these versions:
```
(venv) user@c1b4f29d3e65:~/app$ pip freeze | grep grpcio
grpcio==1.65.5
grpcio-status==1.62.3

(venv) user@c1b4f29d3e65:~/app$ /home/user/venv/bin/python /home/user/app/src/manage.py runserver 0.0.0.0:8080
WARNING: All log messages before absl::InitializeLog() is called are written to STDERR
I0000 00:00:1724192995.212861   13858 check_gcp_environment.cc:61] BIOS data file does not exist or cannot be opened.
...
```

**Comment by Pravitha-o-p** [comment]
> Linked with #37178? I see a similar log in `1.65.4` on Python 3.10 and MacOS (Apple silicon):
> 
> ```
> WARNING: All log messages before absl::InitializeLog() is called are written to STDERR
> I0000 00:00:1723666552.540596 44436952 check_gcp_environment_no_op.cc:29] ALTS: Platforms other than Linux and Windows are not supported
> ```
> 
> Edit: should I open a new issue given that my log is different?

any update on this ?

**Comment by sebastian-correa** [comment]
I goofed up when testing this on my machine as I retested and I still get the logs:

```
WARNING: All log messages before absl::InitializeLog() is called are written to STDERR
I0000 00:00:1724244093.699149 48806386 check_gcp_environment_no_op.cc:29] ALTS: Platforms other than Linux and Windows are not supported
```

on

```bash
❯ pip freeze | grep grpcio
grpcio==1.65.5
grpcio-status==1.62.3
```

**Comment by yashykt** [comment]
> Would we need to backport this to 1.65 too?

1.66 just got released with that PR in it.

**Comment by sourabhsinghs** [comment]
@sshishov  Can you upgrade to 1.66.0 and verify if Issue is indeed fixed for you.

**Comment by sshishov** [comment]
Hi @sourabhsinghs , I can confirm that version `1.66.0` has fixed it for us. No more unnecessary logs shown.

Closing as fixed in v1.66.0 🎉
